### PR TITLE
warning with goto_state in background problem

### PR DESCRIPTION
**Pull request by marcin-usielski**

### CODE REVIEW

File: moler/device/textualdevice.py [hunk 1]
```diff
@@ -316,16 +316,19 @@ def _set_state(self, state):
...
                                 send_enter_after_changed_state=False, log_stacktrace_on_fail=False, keep_state=True)
-            except DeviceChangeStateFailure:
+            except Exception as ex:
                 level = logging.DEBUG
                 if not self._warning_was_sent:
                     level = logging.WARNING
                     self._warning_was_sent = True
-                self._log(level=level, msg="Cannot properly go to state: '{}' in background.".format(state))
-                self._kept_state = state
+                self._log(level=level, msg="Cannot properly go to state: '{}' in background with excprtion.".format(
```
Comment (bodziom777) [inline moler/device/textualdevice.py]: I guess you mean 'exception' instead 'excprtion'

File: moler/device/textualdevice.py [hunk 2]
```diff
@@ -316,16 +316,19 @@ def _set_state(self, state):
...
+                    state, ex))
```
Comment (bodziom777) [inline moler/device/textualdevice.py]: maybe repr(ex) will be more clearly?

File: moler/device/textualdevice.py (hunk 1 above)
Comment (marcin-usielski) [reply moler/device/textualdevice.py]: Done

File: moler/device/textualdevice.py (hunk 2 above)
Comment (marcin-usielski) [reply moler/device/textualdevice.py]: Done

**Pull request merged**